src/votekit/models.py

```diff
@@ -1,95 +1,292 @@
+            round_number = len(self)
 
-    def reset(self):
+        # reverses order to match ranking convention
+        return [
+            s
+            for state in self.election_states[round_number::-1]
+            for s in state.eliminated[::-1]
+            if state.eliminated != (frozenset(),)
+        ]
+
+    def get_remaining(self, round_number: int = -1) -> list[frozenset[str]]:
         """
-        Reset the ElectionState object to initial conditions.
+        Fetch the remaining candidates after the given round.
+
+        Args:
+            round_number (int, optional): Round number, defaults to -1 which corresponds to
+                the final round.
+
+        Returns:
+            list[frozenset[str]]: List of sets of remaining candidates. Ordering of tuple
+            denotes ranking of remaining candidates, sets denote ties.
         """
+        if round_number < -1:
+            raise ValueError("round_number must be -1 or non-negative.")
 
-        self.state = ElectionState(curr_round=0, profile=self._profile)
+        if round_number == -1:
+            round_number = len(self)
 
-    def run_to_step(self, step: int):
+        return list(self.election_states[round_number].remaining)
+
+    def get_ranking(self, round_number: int = -1) -> list[frozenset[str]]:
         """
-        Run the election to the given step.
+        Fetch the ranking of candidates after a given round.
 
         Args:
-            step (int): The step of the election to run to.
+            round_number (int, optional): Round number, defaults to -1 which corresponds to
+                the final round.
 
         Returns:
-            The ElectionState object for round = step.
+            list[set[str]]: Ranking of candidates.
         """
-        if isinstance(step, int):
-            if step < 0:
-                raise ValueError("Step must be a non-negative integer.")
+        if round_number < -1:
+            raise ValueError("round_number must be -1 or non-negative.")
+
+        if round_number == -1:
+            round_number = len(self)
+
+        # len condition handles empty remaining candidates
+        return [
+            s
+            for s in self.get_elected(round_number)
+            + self.get_remaining(round_number)
+            + self.get_eliminated(round_number)
+            if len(s) != 0
+        ]
+
+    def get_status_df(self, round_number: int = -1) -> pd.DataFrame:
+        """
+        Yield the status (elected, eliminated, remaining) of the candidates in the given round.
+        DataFrame is sorted by current ranking.
+
+        Args:
+            round_number (int, optional): Round number, defaults to -1 which corresponds to
+                the final round.
+        Returns:
+            pd.DataFrame:
+                Data frame displaying candidate, status (elected, eliminated,
+                remaining), and the round their status updated.
+        """
+        if round_number < -1:
```

Comment:
Here you can do the same modular indexing thing that I describe in the `get_profile` function